### ISSUE TITLE
OpenWrt detection is faulty for custom builds

### ISSUE THREAD

**Issue by adworacz** [AdguardTeam/AdGuardHome#3435]
I just ran into this bug using a custom build from [Divested](https://forum.openwrt.org/t/divested-wrt-no-nonsense-hardened-builds-for-linksys-wrt-series/80522), which is effectively running OpenWRT HEAD with a few patches.

Looking at the AdguardHome code, it's [checking for "OpenWrt" in `/etc/*release`](https://github.com/AdguardTeam/AdGuardHome/blob/28f34ca399ab27d4780fab6b53a55af8fd7f408f/internal/aghos/os_linux.go#L32-L65).

However `/etc/openwrt_release` DOES exist, and contains the following:
```
DISTRIB_ID='Divested-WRT'
DISTRIB_RELEASE='SNAPSHOT'
DISTRIB_REVISION='r17276+9-b1bff5cb57'
DISTRIB_TARGET='mvebu/cortexa9'
DISTRIB_ARCH='arm_cortex-a9_vfpv3-d16'
DISTRIB_DESCRIPTION='Divested-WRT SNAPSHOT r17276+9-b1bff5cb57'
DISTRIB_TAINTS='no-all'
```

Edit: Actually `/etc/os-release` exists as well, and contains:
```
NAME="Divested-WRT"
VERSION="SNAPSHOT"
ID="divested-wrt"
ID_LIKE="lede openwrt"
PRETTY_NAME="Divested-WRT SNAPSHOT"
VERSION_ID="snapshot"
HOME_URL="https://divested.dev/unofficial-openwrt-builds/mvebu-linksys"
BUG_URL="https://github.com/divestedcg/Divested-WRT/issues"
SUPPORT_URL="xmpp:divested-wrt@conference.konvers.me?join"
BUILD_ID="r17276+9-b1bff5cb57"
OPENWRT_BOARD="mvebu/cortexa9"
OPENWRT_ARCH="arm_cortex-a9_vfpv3-d16"
OPENWRT_TAINTS="no-all"
OPENWRT_DEVICE_MANUFACTURER="Divested Computing Group"
OPENWRT_DEVICE_MANUFACTURER_URL="https://divested.dev"
OPENWRT_DEVICE_PRODUCT="Generic"
OPENWRT_DEVICE_REVISION="v0"
OPENWRT_RELEASE="Divested-WRT SNAPSHOT r17276+9-b1bff5cb57"
```

**In short - the detection is looking for the "OpenWrt" string (case sensitive), while there are `*release` files with lowercase "openwrt" strings.**

_Originally posted by @adworacz in https://github.com/AdguardTeam/AdGuardHome/issues/2757#issuecomment-894702938_

**Comment by EugeneOne1** [comment]
@adworacz, latest build in the edge channel should fix the issue.  Could you please install it and check OpenWrt detection?

**Comment by adworacz** [comment]
Yup I’ll give it a shot this weekend!

**Comment by adworacz** [comment]
Works like a champ! 

The correct init script is installed, and the service starts correctly on reboot. Thank you for the fix!

**Comment by EugeneOne1** [comment]
Glad to hear that.  We'll close the issue for now.
